core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -189,6 +189,20 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertThrows(classOf[InvalidTopicException], () => consumer.partitionsFor(";3# ads,{234"))
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
+  def testSubscribeInvalidTopic(quorum: String, groupProtocol: String): Unit = {
+    val invalidTopicName = "topic abc"
+    val consumer = createConsumer()
+
+    // subscribe invalid topic
+    consumer.subscribe(List(invalidTopicName).asJava)
+
+    consumer.poll(Duration.ZERO);
```

Comment:
Without this line, it will fail when groupProtocol is consumer.
When debugging, after processing the Background Event, Consumer receives an `InvalidTopicException`.
So adding `consumer.poll(Duration.ZERO)`

Do you have any thoughts on this? @philipnee & @lianetm

Reply:
actually, looking at the code, it seems like `updateAssignmentMetadataIfNeeded` should continue to poll the background queue, i think i'm missing something.